### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by BeyondEvil**

### CODE REVIEW

File: src/pytest_html/html_report.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
+import bisect
+import datetime
+import json
+import os
+import re
+import time
+from collections import defaultdict
+from collections import OrderedDict
+
+from py.xml import html
```
Comment (RonnyPfannschmidt) [inline src/pytest_html/html_report.py:10]: can we get rid of that somehow, pylib is "dead"

File: src/pytest_html/html_report.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        for key in keys:
+            value = metadata[key]
+            if self._is_redactable_environment_variable(key, config):
+                black_box_ascii_value = 0x2593
+                value = "".join(chr(black_box_ascii_value) for char in str(value))
+
+            if isinstance(value, str) and value.startswith("http"):
+                value = html.a(value, href=value, target="_blank")
+            elif isinstance(value, (list, tuple, set)):
+                value = ", ".join(str(i) for i in sorted(map(str, value)))
```
Comment (RonnyPfannschmidt) [inline src/pytest_html/html_report.py:237]:
```suggestion
                value = ", ".join(sorted(str(i) for i in value))
```

File: src/pytest_html/html_report.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+            elif isinstance(value, dict):
+                sorted_dict = {k: value[k] for k in sorted(value)}
```
Comment (RonnyPfannschmidt) [inline src/pytest_html/html_report.py:239]:
```suggestion
                value = json.dumps(value,  sort_keys=True)
```

a future addon might be to have a interactive intended view of the data that can expand/fold

File: src/pytest_html/html_report.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+                value = json.dumps(sorted_dict)
+            raw_value_string = raw(str(value))
+            rows.append(html.tr(html.td(key), html.td(raw_value_string)))
+
+        environment.append(html.table(rows, id="environment"))
+        return environment
+
+    def _is_redactable_environment_variable(self, environment_variable, config):
+        redactable_regexes = config.getini("environment_table_redact_list")
```
Comment (RonnyPfannschmidt) [inline src/pytest_html/html_report.py:248]: potential future enhancement - compiling a combined pattern via https://docs.python.org/3/library/re.html#writing-a-tokenizer and using it  to match repeatedly (needs perf test to be sure)

File: src/pytest_html/html_report.py (hunk 1 above)
Comment (BeyondEvil) [reply src/pytest_html/html_report.py:10]: Hi @RonnyPfannschmidt 

Yes, the point of "Next Gen" is to get rid of that, and other things. So basically, this whole file will be replaced, if not in its entirety, then a lot of it.

I appreciate you taking a look, but this is far from ready. I should've set it to "Draft". Apologies 🙏

File: src/pytest_html/result.py [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+            log.append(f" {header:-^80} ")
+            log.append(html.br())
+
+            if ansi_support():
+                converter = ansi_support().Ansi2HTMLConverter(
+                    inline=False, escaped=False
+                )
+                content = converter.convert(content, full=False)
+            else:
+                content = _remove_ansi_escape_sequences(content)
```
Comment (RonnyPfannschmidt) [inline src/pytest_html/result.py:208]: this block reads like something to go to the ansi module itself - it also looks to melike the import will always try, the module itself can change

File: src/pytest_html/html_report.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/pytest_html/html_report.py:10]: i was curious anyway ^^ i'd love to have a little talk/chat about this sometime soon

Comment (BeyondEvil) [reply src/pytest_html/html_report.py:10]: Let me get it to a workable state, and I'm happy to chat @RonnyPfannschmidt! I know @davehunt wants to join as well.

**Pull request merged**